src/Components/Components/src/DefaultComponentActivator.cs

```diff
@@ -10,15 +11,14 @@ internal class DefaultComponentActivator : IComponentActivator
         public static IComponentActivator Instance { get; } = new DefaultComponentActivator();
 
         /// <inheritdoc />
-        public IComponent CreateInstance(Type componentType)
+        public IComponent CreateInstance([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type componentType)
         {
-            var instance = Activator.CreateInstance(componentType);
-            if (!(instance is IComponent component))
+            if (!typeof(IComponent).IsAssignableFrom(componentType))
```

Comment:
Is the change here in order to improve things for linkability, or just a stylistic preference?

It's probably not a big deal but the implementation before avoided repeated type compatibility checks.

Reply:
I think it might allow the linker to better statically analyze the code particularly if it's unable to piece together the control flow that this is a component type. 

Do you think it's worthwhile removing the type compatibility check (or putting it behind a `Debug.Assert` if it helps the linker analysis) and letting the cast throw? The exception message you get for the invalid cast isn't significantly worse than what you currently get.

Reply:
It could help analysis, but illink doesn't work that way right now.

The purpose is to ensure that if someone passes a random type here, we don't get a MissingMemberException due to trimming before we would do the type check.

The behavior of the app before/after trimming shouldn't change and that includes error cases like this - they make troubleshooting harder.

Reply:
> Do you think it's worthwhile removing the type compatibility check (or putting it behind a Debug.Assert if it helps the linker analysis) and letting the cast throw? The exception message you get for the invalid cast isn't significantly worse than what you currently get.

I'm certainly fine with letting the error come from the cast if that gives us some linkability benefit. I agree the message on a failed cast exception is clear enough.